test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,433 @@
+        box_cfg = {
+            replication = {
+                server.build_listen_uri('replica', cg.cluster.id),
+                server.build_listen_uri('master_proxy'),
+            },
+            election_mode = 'off',
+            replication_synchro_quorum = 2,
+            replication_synchro_timeout = 100000,
+        }
+    })
+    cg.master_proxy = proxy:new({
+        client_socket_path = server.build_listen_uri('master_proxy'),
+        server_socket_path = server.build_listen_uri('master', cg.cluster.id),
+    })
+    cg.master_proxy:start({force = true})
+    cg.replica_proxy = proxy:new({
+        client_socket_path = server.build_listen_uri('replica_proxy'),
+        server_socket_path = server.build_listen_uri('replica', cg.cluster.id),
+    })
+    cg.replica_proxy:start({force = true})
+    cg.cluster:start()
+    cg.master:wait_until_election_leader_found()
+    cg.replica:wait_until_election_leader_found()
+
+    cg.master:exec(function()
+        box.schema.space.create('test', {is_sync = true})
+        box.space.test:create_index('pk')
+    end)
+    wait_pair_sync(cg.replica, cg.master)
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_wal_sync_fails = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_SYNC', true)
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_geq_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function(f)
+        t.assert_equals(box.info.synchro.queue.len, 1)
+        box.error.injection.set('ERRINJ_WAL_SYNC', false)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        require('fiber').find(f):join()
+    end, {f})
+    server_becomes_the_leader_again(cg.master)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_update_greatest_term_while_wal_sync = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    server_becomes_the_leader_again(cg.master)
```

Comment:
I don't understand why this bumps 'promote greatest term' on the replica. AFAIU replica is still blocked in writing new Term, and the master is the leader with `election_mode='candidate'` anyway. So this call won't bump the term and result in "promote".

I think it would be easier to make the master bump term by first setting `election_mode='off'` and then writing `box.ctl.demote()` on it.